src/main/kotlin/org/jetbrains/intellij/platform/gradle/models/IvyModule.kt

```diff
@@ -88,10 +100,79 @@ data class IvyModule(
  *
  * @see IntelliJPlatformRepositoriesExtension.jetbrainsIdeInstallers
  */
-internal fun Path.toIvyArtifact() = IvyModule.Artifact(
-    name = invariantSeparatorsPathString.replaceFirst(Regex("^[a-zA-Z]:/"), "/"),
-    type = when {
-        isDirectory() -> "directory"
-        else -> extension
-    },
-)
+internal fun Path.toIvyArtifact(): IvyModule.Artifact {
+    val alwaysEmpty = ""
+    val trimmedAbsolutePath = this.removeLeadingPathSeparator()
+    val ext = when {
+        this.isDirectory() -> "directory"
+        else -> this.extension
+    }
+
+    // In this case, name is the full path, so type is not needed. It can be empty because in:
+    // IntelliJPlatformRepositoriesHelper.createLocalIvyRepository the pattern is "/([type])[artifact]" where the type
+    // is marked as optional.
+    // In this case, it should be ok to have an absolute path in the name, because this is a fallback method,
+    // which called for directories only (at least at the moment of writing this).
+    // See comments in explodeIntoIvyJarsArtifacts on why having abs path in name may be bad.
+    return IvyModule.Artifact(type = alwaysEmpty, name = trimmedAbsolutePath, ext = ext)
+}
+
+internal fun Path.toBundledPluginIvyArtifacts(): List<IvyModule.Artifact> {
+    return this.explodeIntoIvyJarsArtifacts()
+}
+
+internal fun Path.toBundledModuleIvyArtifacts(): List<IvyModule.Artifact> {
+    return this.explodeIntoIvyJarsArtifacts()
+}
+
+internal fun Path.toLocalPluginIvyArtifacts(): List<IvyModule.Artifact> {
+    return this.explodeIntoIvyJarsArtifacts()
+}
+
+/** This method is a bit too universal. It could be split into 3 separate with unnecessary logic removed. */
```

Comment:
I left the comment but did not change it to avoid making this PR too big. I implemented this similarly to how it was wording before.